src/m365/commands/login.ts

```diff
@@ -199,32 +203,36 @@ class LoginCommand extends Command {
         auth.connection.cloudType = CloudType.Public;
       }
 
-      try {
-        await auth.ensureAccessToken(auth.defaultResource, logger, this.debug);
-        auth.connection.active = true;
-      }
-      catch (error: any) {
-        if (this.debug) {
-          await logger.logToStderr('Error:');
-          await logger.logToStderr(error);
-          await logger.logToStderr('');
-        }
-
-        throw new CommandError(error.message);
-      }
+      await this.obtainAccessToken(logger);
 
       const details = auth.getConnectionDetails(auth.connection);
 
       if (this.debug) {
         (details as any).accessToken = JSON.stringify(auth.connection.accessTokens, null, 2);
       }
 
-
       await logger.log(details);
     };
 
-    deactivate();
-    await login();
+    try {
+      if (this.shouldRenewConnection(args.options)) {
+        await auth.clearConnectionInfo();
+      }
+    }
+    catch (error: any) {
+      if (this.debug) {
+        await logger.logToStderr(new CommandError(error));
+      }
+    }
+    finally {
+      if (!this.shouldRenewConnection(args.options)) {
```

Comment:
We've already checked if we should renew the connection on line 218 so let's avoid repeating it and instead, check it once, and store the result so that we can reuse it.